newIDE/app/src/ProjectManager/ProjectManagerItems.js

```diff
@@ -0,0 +1,304 @@
+                },
+                { type: 'separator' },
+                {
+                  label: i18n._(t`Copy`),
+                  click: () => onCopy(),
+                },
+                {
+                  label: i18n._(t`Cut`),
+                  click: () => onCut(),
+                },
+                {
+                  label: i18n._(t`Paste`),
+                  enabled: canPaste(),
+                  click: () => onPaste(),
+                },
+                {
+                  label: i18n._(t`Duplicate`),
+                  click: () => onDuplicate(),
+                },
+                { type: 'separator' },
+                {
+                  label: i18n._(t`Move up`),
+                  enabled: canMoveUp,
+                  click: () => onMoveUp(),
+                },
+                {
+                  label: i18n._(t`Move down`),
+                  enabled: canMoveDown,
+                  click: () => onMoveDown(),
+                },
+              ]}
+              onClick={() => {
+                // It's essential to discard clicks when editing the name,
+                // to avoid weird opening of an editor (accompanied with a
+                // closing of the project manager) when clicking on the text
+                // field.
+                if (!editingName) onEdit();
+              }}
+            />
+          )}
+        </I18n>
+      )}
+    </ThemeConsumer>
+  );
+};
+
+type EventFunctionExtensionItemProps = {|
+  eventsFunctionsExtension: gdEventsFunctionsExtension,
+  onEdit: ({ [string]: ExtensionShortHeader }) => void,
+  onRename: string => void,
+  onEditName: () => void,
+  isEditingName: boolean,
+  onDelete: () => void,
+  onAdd: () => void,
+  onCopy: () => void,
+  onCut: () => void,
+  onPaste: () => void,
+  onDuplicate: () => void,
+  canPaste: () => boolean,
+  canMoveUp: boolean,
+  onMoveUp: () => void,
+  canMoveDown: boolean,
+  onMoveDown: () => void,
+|};
+
+export const EventFunctionExtensionItem = ({
+  eventsFunctionsExtension,
+  onEdit,
+  onRename,
+  onEditName,
+  isEditingName,
+  onDelete,
+  onAdd,
+  onCopy,
+  onCut,
+  onPaste,
+  onDuplicate,
+  canPaste,
+  canMoveUp,
+  onMoveUp,
+  canMoveDown,
+  onMoveDown,
+}: EventFunctionExtensionItemProps) => {
+  const name = eventsFunctionsExtension.getName();
+  const iconUrl = eventsFunctionsExtension.getIconUrl();
+
+  const { extensionShortHeadersByName } = React.useContext(
+    ExtensionStoreContext
+  );
+
+  return (
+    <Item
+      leftIcon={
+        iconUrl ? (
+          <IconContainer
+            size={24}
+            alt={eventsFunctionsExtension.getFullName()}
+            src={iconUrl}
+          />
+        ) : null
+      }
+      primaryText={name}
+      editingName={isEditingName}
+      onEdit={() => onEdit(extensionShortHeadersByName)}
+      onDelete={onDelete}
+      addLabel={t`Add a New Extension`}
+      onAdd={onAdd}
+      onRename={newName => {
```

Comment:
this can just be `onRename={onRename}`?

Reply:
Thanks for spotting it, I had many versions and my eyes were tired.